lib/bodyParser.js

```diff
@@ -0,0 +1,31 @@
+const { parse } = require('querystring');
+
+module.exports = (routing, pageNotFound = 'page not found') => (
+  req,
+  res,
+  next
+) => {
+  if (req.method === 'POST' || req.method === 'PUT') {
+    let method = req.method.toLowerCase();
+    if (routing[method][req.url]) {
+      let body = '';
+
+      req.on('data', function (data) {
+        body += data;
+      });
+
+      req.on('end', function () {
+        try {
+          req.body = JSON.parse(body);
+        } catch (e) {
+          req.body = parse(body);
+        }
+      });
```

Comment:
See this: https://github.com/metarhia/metacom/blob/f643f4d42aaba954b08e6b2c86f8105cb13fd69f/lib/server.js#L15-L21